### DISCUSSION TITLE
Purpose of the SSL/TLS switch

### DISCUSSION THREAD

**Discussion by AndreKR** [emqx/MQTTX#1087] [Q&A]
There is a toggle switch in the connection form that's labeled "SSL/TLS":

![image](https://user-images.githubusercontent.com/1188538/194138114-6bbeb0a3-e89b-41b1-8ae5-849b2cb178eb.png)

What it seems to do is to show an "SSL" corner next to the connection:

![image](https://user-images.githubusercontent.com/1188538/194138266-7d7ddcc2-1f3f-4e80-995c-ee7a6b11e828.png)

Is that really all it does?

The connection itself seems to be always made using TLS as long as "mqtts://" is chosen as the protocol.

Another question: Is the certificate host name verified when the toggle switch is off?

**Comment by ysfscream** [comment]
Hello, thank you for your feedback!

You are right, at the moment it seems that if you don't turn on the `ssl/tls` switch and select `mqtts` or wss protocol, you seem to be able to connect correctly. Here is actually a logical error, this switch is really to let the user know that when opening `ssl/tls`, you need to switch the protocol, if it is self-signed, you also need to manually upload the certificate, but if it is CA signed server, it may not be necessary, I will fix this logical flaw in the follow-up
